crates/router/src/connector/zen/transformers.rs

```diff
@@ -395,15 +395,20 @@ fn get_item_object(
     _amount: String,
 ) -> Result<Vec<ZenItemObject>, error_stack::Report<errors::ConnectorError>> {
     let order_details = item.request.get_order_details()?;
-    Ok(order_details
+
+    order_details
         .iter()
-        .map(|data| ZenItemObject {
-            name: data.product_name.clone(),
-            quantity: data.quantity,
-            price: data.amount.to_string(),
-            line_amount_total: (i64::from(data.quantity) * data.amount).to_string(),
+        .map(|data| {
+            Ok(ZenItemObject {
+                name: data.product_name.clone(),
+                quantity: data.quantity,
+                price: utils::to_currency_base_unit(data.amount, item.request.currency)?,
```

Comment:
Assign 'utils::to_currency_base_unit(data.amount, item.request.currency)?' this to a variable and reuse it here and in line number 407

Reply:
Both are different function calls, First one returns a string, the second one returns an f64 which is further processed

Reply:
Ignore this